src/ios-scroll.js

```diff
@@ -0,0 +1,557 @@
+   *
+   * @param {!Element} element
+   */
+  removeFixedElement(element) {
+    const style = element.style;
+    const index = this.fixedElementStyles_.indexOf(style);
+
+    if (index > -1) {
+      this.fixedElementStyles_.splice(index, 1);
+      style.transform = '';
+    }
+  }
+
+  /**
+   * Adds an iframe to the list we track. When the doc is scrolled, we will
+   * disable interaction with the element to prevent native scrolling from
+   * kicking in. When scrolling has stopped, we will re-enable.
+   *
+   * @param {!HTMLIFrameElement} iframe
+   */
+  addIframe(iframe) {
+    const style = iframe.style;
+    this.iframeStyles_.push(style);
+
+    // Offset is only present during scrolling, in which case we need to
+    // disable interacting with the iframe.
+    if (this.offset) {
+      style.pointerEvents = 'none';
+    }
+  }
+
+  /**
+   * Removes the iframe from our list.
+   *
+   * @param {!HTMLIFrameElement} iframe
+   */
+  removeIframe(iframe) {
+    const style = iframe.style;
+    const index = this.iframeStyles_.indexOf(style);
+
+    if (index > -1) {
+      this.iframeStyles_.splice(index, 1);
+      style.pointerEvents = '';
+    }
+  }
+
+  /**
+   * Scrolls the document to position ypos.
+   *
+   * @param {number} ypos
+   */
+  setScrollTop(ypos) {
+    this.scrollTo_(ypos);
+    this.scrollStopped_();
+  }
+
+  /**
+   * Calculates the current scrollTop.
+   */
+  getScrollTop() {
+    const delta = this.bodyScrolled_ ?
+        -this.scrollMoveEl_./*OK*/getBoundingClientRect().top  :
+        0;
+    return this.scrollTop_ + this.offset_ + delta;
+  }
+
+  /**
+   * Resize the scroll element. Usually triggered after an orientation change.
+   *
+   * @param {number} scrollTop
+   * @param {number} scrollHeight
+   */
+  resize(scrollTop, scrollHeight) {
+    var delta = this.scrollMoveEl_./*OK*/getBoundingClientRect().top;
+    this.scrollTop_ = scrollTop + delta;
+    this.min_ = -this.scrollTop_;
+    this.max_ = Math.round(scrollHeight - this.win_./*OK*/innerHeight) - delta;
+  }
+
+  /**
+   * Registers the handler for scroll events.
+   *
+   * @param {!function()} handler
+   * @return {!Unlisten}
+   */
+  onScroll(handler) {
+    return this.scrollObservable_.add(handler);
+  }
+
+  /**
+   * Synthetically scrolls the document to position ypos. This does not change
+   * the body's scrollTop.
+   *
+   * @param {number} ypos
+   * @private
+   */
+  scrollTo_(yPos) {
+    const y = Math.min(Math.max(yPos, this.min_), this.max_);
+    if (y === this.offset_) {
+      return false;
+    }
+    const scrollEventDelta = Math.abs(this.lastScrollEventOffset_ - y);
+    this.offset_ = y;
+
+    // "Scroll" by moving the scroller element in the opposite direction of
+    // the scroll.
+    // We remove the transform when there is no offset so that the body may be
+    // sync'd with our synthetic scroll position.
+    this.scrollStyle_.style.transform = y ? `translate3d(0, ${-y}px, 0)` : '';
```

Comment:
Whoot, you accidentally caught an error! But, `setStyle` will actually be a slowdown here:

- iOS Safari back to (at least) iOS 8 supports the `transform` style. I could directly use the `webkitTransform` style, though.
- We're storing the `CSSStyleDeclaration` directly to avoid the expensive lookup.